### PR TITLE
teo/feat/add page setup uno commands

### PR DESCRIPTION
**Pull request by synoet**
Adds uno commands: [`.uno:SetPageColor`, `.uno:ToggleOrientation`, `.uno:SetPageSize`, `.uno:SetPageMargins`]
Adds command valuesL [`.uno:PageColor`, `.uno:Orientation`, `.uno:PageSize`, `.uno:PageMargins`]

https://www.loom.com/share/822b1104badb46118f51b4ce189b85f0

### CODE REVIEW

File: libreoffice-core/include/LibreOfficeKit/LibreOfficeKit.h [hunk 1]
```diff
@@ -245,6 +245,9 @@ struct _LibreOfficeKitDocumentClass
                                const char* pMimeType,
                                char** pUsedMimeType);
 
+    char* (*getPageMargins) (LibreOfficeKitDocument* pThis);
+    char* (*getPageSize) (LibreOfficeKitDocument* pThis);
```
Comment (chase) [inline libreoffice-core/include/LibreOfficeKit/LibreOfficeKit.h:249]: You can avoid this by extending using getCommandValues.

File: libreoffice-core/desktop/source/lib/init.cxx [hunk 2]
```diff
@@ -889,6 +892,49 @@ void ExecuteMarginULChange(
...
+)
+{
+    SfxViewFrame* pViewFrm = SfxViewFrame::Current();
+
+    if (!pViewFrm)
+        return;
+
+    SvxLongLRSpaceItem* pLRSpaceItem = new SvxLongLRSpaceItem();
+    pLRSpaceItem->SetLeft(PageLeft);
+    pLRSpaceItem->SetRight(PageRight);
```
Comment (chase) [inline libreoffice-core/desktop/source/lib/init.cxx:927]: Best way to validate this is to is correct is to save the document, and pop it into LibreOffice and seeing if the values are the right ones

File: libreoffice-core/vs-code-template.code-workspace
Comment (chase) [inline libreoffice-core/vs-code-template.code-workspace:1]: It says autogenerated but this is actually the manually corrected version I added, fwiw

File: libreoffice-core/desktop/source/lib/init.cxx [hunk 3]
```diff
@@ -3498,6 +3603,9 @@ static void doc_iniUnoCommands ()
         OUString(".uno:Watermark"),
         OUString(".uno:ResetAttributes"),
         OUString(".uno:Orientation"),
+        OUString(".uno:SetPageMargins"),
+        OUString(".uno:SetPageColor"),
+        OUString(".uno:GetPageMargins"),
```
Comment (chase) [inline libreoffice-core/desktop/source/lib/init.cxx:3608]: At the very, very bottom of this function you'll see this:

```
    SfxSlotPool& rSlotPool = SfxSlotPool::GetSlotPool(pViewFrame);
    uno::Reference<util::XURLTransformer> xParser(util::URLTransformer::create(xContext));

    for (const auto & sUnoCommand : sUnoCommands)
    {
        aCommandURL.Complete = sUnoCommand;
        xParser->parseStrict(aCommandURL);

        // when null, this command is not supported by the given component
        // (like eg. Calc does not have ".uno:DefaultBullet" etc.)
        if (const SfxSlot* pSlot = rSlotPool.GetUnoSlot(aCommandURL.Path))
        {
            // Initialize slot to dispatch .uno: Command.
            pViewFrame->GetBindings().GetDispatch(pSlot, aCommandURL, false);
        }
    }
```

Basically, this is only for uno commands that are translated to an sfx slot, so something with a `SID_`

File: libreoffice-core/desktop/source/lib/init.cxx [hunk 4]
```diff
@@ -8,8 +8,11 @@
  */
 
 #include "comphelper/seqstream.hxx"
+#include "editeng/editids.hrc"
+#include "editeng/formatbreakitem.hxx"
 #include "lib/unov8.hxx"
 #include <sal/backtrace.hxx>
+#include "sfx2/QuerySaveDocument.hxx"
```
Comment (chase) [inline libreoffice-core/desktop/source/lib/init.cxx:15]: I'm guessing you didn't add these intentionally. Are these required?

Comment (synoet) [reply libreoffice-core/desktop/source/lib/init.cxx]: Sorry yes removed the ones that where not necessary.

Comment (synoet) [reply libreoffice-core/desktop/source/lib/init.cxx]: I think clangd added these automatically while I was playing around with some stuff.

**Review by chase — APPROVED** [review]

**Pull request merged**